# Variables Research (Part IV - Sales Related Variables)

### `Misael Ramirez - A00821781`

<img src="autlan-logo.png" alt="Autlan logo">


In this notebook, we will perform a research for a financial forecast of variables which driver is directly related with sales. Sales are divided in the following:
1. *Mining-Metallurgical Manganese Products*
2. *Energy*
3. *Precious Metals*

Nonetheless, the contribution of the **energy** division is technically none, since the resources related with energy are used in the mines possesed by Autlán to perform the extraction of minerals, manganese for the most part. As stated in 2Q14 report: 
> "The electricity generation of the Atexcaco hydroelectric plant was favored by the constant flow of water during the dry months, so its generation grew 56%, contributing 30% of Autlán's needs and representing savings in the first half of the year of 40.2 million MXN"

**Source:** Autlan

Now, we will focus on variables that use sales as a driver too, but this are more related with the way Autlán operates based on the forecasted sales.

In [4]:
# import required libraries 
import seaborn as sns
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib as mpl

# setting up future plots
sns.set_style(  
    "darkgrid",  
    {  
        'legend.frameon': True,  
    }  
)
mpl.rc('figure', figsize=(14, 8))

## Forecast Cost of Gods Sold (2020-2025)

We perform a linear regression with the power of machine learning, dividing our data for trainign and the prediction itself. The COGS for FY20 will be calculated with the Sales from that same year we estimated through our research. 

The way we calculate gradients will be different, instead of modeling the next monetary quantity of COGS for the next years we will focus on the gradients directly for the forecast model. The percentage of COGS does not vary a lot over time (not even the gradient for FY20 which comes from forecasted numbers, COGS and Sales respectively), and this can be seen in its variance over time, which is `5.2415`, considering that we are considering millions of dollars, it doesn't change as much.

We will use the following logic for the gradients

>**The years that the team considered of economic slowdown (FY20-FY22) will show a constat gradient of `plus 2 percent` in COGS until arriving to the years of economic recovery where COGS will show a constant gradient of `minus 2 percent`**

This results in variables show a tendency very similar to the previous five years of analysis

In [28]:
# import required libraries
from sklearn.linear_model import LinearRegression 

# create data for model
data = pd.read_excel('other_vars.xlsx', sheet_name='COGS_1')
data = data.drop('Date', axis=1)

Y_train = data['COGS'].values.reshape(-1, 1)
X_train = data['Sales'].values.reshape(-1, 1)
X_test = np.array([320.35]).reshape(-1, 1)

print('COSG FY2014-FY2019: \n{}\n'.format(Y_train))
print('Sales FY2014-FY2019:\n{}\n'.format(X_train))
print('Predicted Sales FY20: {}\n'.format(X_test))

# create model
reg = LinearRegression()
reg.fit(X_train,Y_train)
y_pred = reg.predict(X_test)
print('COGS FY20: {}'.format(y_pred))

COSG FY2014-FY2019: 
[[244.723]
 [189.118]
 [152.038]
 [196.723]
 [275.567]
 [285.362]]

Sales FY2014-FY2019:
[[338.04 ]
 [267.729]
 [230.711]
 [359.34 ]
 [413.504]
 [420.128]]

Predicted Sales FY20: [[320.35]]

COGS FY20: [[212.62949882]]


In [26]:
print('Standard Deviation: {}'.format(np.std([72.39, 70.64, 65.90, 54.75, 66.64, 67.92, 66])))
data = pd.read_excel('other_vars.xlsx', sheet_name='COGS_2')
display(data)

Standard Deviation: 5.241501965767337


Date       Sales        COGS         %
0   2014-12-31  338.040000  244.723000  0.723900
1   2015-12-31  267.729000  189.118000  0.706400
2   2016-12-31  230.711000  152.038000  0.659000
3   2017-12-31  359.340000  196.723000  0.547500
4   2018-12-31  413.504000  275.567000  0.666400
5   2019-12-31  420.128000  285.362000  0.679200
6   2020-12-31  320.350000  212.629000  0.663740
7   2021-12-31  317.909690  216.881580  0.682211
8   2022-12-31  315.851145  221.219212  0.700391
9   2023-12-31  325.756420  216.794827  0.665512
10  2024-12-31  353.382093  212.458931  0.601216
11  2025-12-31  385.469342  208.209752  0.540146

## Forecast of General Expenses (2020-2025)

> Similar low variance gradients such as cost of good sols

> Follow similar gradient logic as COGS

## Capital Expenditure (CAPEX)

Due to the economic uncertainty, unnecesary risks will not be taken, we keep a low CAPEX such as 2016 and 2017, which is before acquiring Metallorum and having a 55% increase in CAPEX.

As a conclusion, we will apply the following logic for the gradients:

>**The forecast will be a low variance gradient such as FY16 and Fy17 with a base gradient of 3% which increases 75% of the previous FY each year as the economic turmoil passes and opens more investment opportunities for autlán and allows to improve the management operations of Metallorum**

## Días Cuentas por Cobrar

> Similar low variance gradients such as cost of good sols

> Follow similar gradient logic as COGS

## Días Inventarios

> Similar low variance gradients such as cost of good sols

> Follow similar gradient logic as COGS

## Días Proveedores

> Similar low variance gradients such as cost of good sols

> Follow similar gradient logic as COGS